metaserver/src/test/scala/tests/index/SymbolIndexerTest.scala

```diff
@@ -0,0 +1,214 @@
+package tests.index
+
+import java.io.ByteArrayInputStream
+import java.io.InputStream
+import java.io.PipedInputStream
+import java.io.PipedOutputStream
+import java.net.URI
+import java.net.URLClassLoader
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+import java.nio.file.Paths
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+import scala.meta.languageserver.ScalametaLanguageServer
+import scala.meta.languageserver.Semanticdbs
+import scala.meta.languageserver.ServerConfig
+import scala.meta.languageserver.SymbolIndexer
+import scala.meta.languageserver.ctags.Ctags
+import scala.meta.languageserver.{index => i}
+import scala.{meta => m}
+import langserver.core.MessageReader
+import langserver.core.MessageWriter
+import langserver.messages.ClientCapabilities
+import langserver.messages.DefinitionResult
+import langserver.messages.TextDocumentDefinitionRequest
+import langserver.messages.TextDocumentPositionParams
+import langserver.types.TextDocumentIdentifier
+import org.langmeta.inputs.Input
+import org.langmeta.internal.io.FileIO
+import org.langmeta.internal.io.PathIO
+import org.langmeta.internal.semanticdb.{schema => s}
+import langserver.{types => l}
+import org.langmeta.io.AbsolutePath
+import org.langmeta.languageserver.InputEnrichments._
+import org.langmeta.semanticdb.Database
+import tests.MegaSuite
+import utest._
+import monix.execution.schedulers.TestScheduler
+import org.langmeta.io.Classpath
+import play.api.libs.json.Format
+import play.api.libs.json.JsValue
+import play.api.libs.json.Json
+
+object SymbolIndexerTest extends MegaSuite {
+  implicit val cwd: AbsolutePath =
+    PathIO.workingDirectory.resolve("test-workspace")
+  val path = cwd
+    .resolve("a")
+    .resolve("src")
+    .resolve("test")
+    .resolve("scala")
+    .resolve("example")
+    .resolve("UserTest.scala")
+  assert(Files.isRegularFile(path.toNIO))
+
+  val scheduler = TestScheduler()
+  val config = ServerConfig(
+    cwd,
+    setupScalafmt = false,
+    indexJDK = false,
+    indexClasspath = true // set to false to speedup edit/debug cycle
+  )
+  val client = new PipedOutputStream()
+  val stdin = new PipedInputStream(client)
+  val stdout = new PipedOutputStream()
+
+  val server = new ScalametaLanguageServer(
+    config,
+    stdin,
+    stdout,
+    System.out
+  )(scheduler)
+  server.initialize(0L, cwd.toString(), ClientCapabilities())
+  // TODO(olafur) run this as part of utest.runner.Framework.setup()
+  while (scheduler.tickOne()) () // Trigger indexing
+  val indexer: SymbolIndexer = server.symbolIndexer
+
+  override val tests = Tests {
+
+    "fallback" - {
+      "<<User>>(...)" - {
+        val term = indexer.findSymbol(path, 3, 17).get
```

Comment:
how does this fail in case a symbol is not found?

Maybe it's worth to also check that `term` is defined?

```scala
val term = indexer.findSymbol(path, 3, 17)
assert(term.isDefined)
assertNoDiff(term.get.symbol, "_root_.a.User#")
assert(term.get.definition.isDefined)
```

Or even a common utility as:

```scala
def assertSymbolFound(path: AbsolutePath, line: Int, column: Int, expected: String) = {
  val term = indexer.findSymbol(path, line, column)
  assert(term.isDefined, s"Symbol not found at $path:$line:$column")
  assertNoDiff(term.get.symbol, expected)
  assert(term.get.definition.isDefined, s"Definition not found for term $term")
}
```